strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/GroupRepositoryProvider.java

```diff
@@ -80,15 +77,22 @@ protected InputStream getInputStreamInternal(RepositoryPath path) throws IOExcep
 
     @Override
     public RepositoryPath fetchPath(RepositoryPath repositoryPath)
-        throws IOException
+            throws IOException
     {
         RepositoryPath result = resolvePathDirectlyFromGroupPathIfPossible(repositoryPath);
-        if (result != null)
+        if (result == null || RepositoryFiles.hasExpired(result))
```

Comment:
I think it will be better to have it like it was 
```
if ( result != null && !RepositoryFiles.hasExpired(result))
```
this will have less complexity, with no nested `if`